pkg/engine/deployment.go

```diff
@@ -152,6 +153,7 @@ type deploymentOptions struct {
 
 // deploymentSourceFunc is a callback that will be used to prepare for, and evaluate, the "new" state for a stack.
 type deploymentSourceFunc func(
+	cancel context.Context,
```

Comment:
I would just call this `ctx`, since presumably this context can be used generically. We are just introducing it to allow cancelation.
```suggestion
	ctx context.Context,
```

If you just want to allow cancelation, I would pass in a `chan<- struct`.